Source/CameraView/CameraView.swift

```diff
@@ -127,7 +127,14 @@ class CameraView: UIViewController {
       }, completion: { finished in
         self.captureSession.beginConfiguration()
         self.captureSession.removeInput(currentDeviceInput)
-        do { try self.captureSession.addInput(AVCaptureDeviceInput(device: self.captureDevice)) } catch {}
+
+        if self.captureDevice!.supportsAVCaptureSessionPreset(AVCaptureSessionPreset1280x720) {
+          self.captureSession.sessionPreset = AVCaptureSessionPreset1280x720
+        } else {
+          self.captureSession.sessionPreset = AVCaptureSessionPreset640x480
+        }
+
+        try! self.captureSession.addInput(AVCaptureDeviceInput(device: self.captureDevice))
```

Comment:
@zenangst yes. If I choose capture session device-specific, it won't fail. If it fails, app should terminate, but this case is impossible. All devices supporting iOS 8+ have camera at least better than 4S. This code works on 4S -> It should work everywhere.